ordering/__init__.py

```diff
@@ -61,6 +61,18 @@ def compare(self, left_item: T, right_item: T) -> bool:
 
         return self._labels[left_item] < self._labels[right_item]
 
+    def replace(self, existing_item: T, new_item: T) -> None:
```

Comment:
Good question, though it'd still need the `self.assert_contains(existing_item)` above. 

My only argument would be to avoid the unnecessary assertions, though I have in mind a way to get around that in general, anyway, if it's important.